### ISSUE TITLE
Permanent search icon in front of megabar has no tooltip/accessibility label and no keyboard functionality

### ISSUE THREAD

**Issue by bkaluzny** [bkaluzny/searchJIRAimport#65]
Just for something new and different 😒, we have a new toolbar button - the permanent search icon introduced in bug 1589836 - which has no tooltip/accessibility label and cannot be reached nor activated via the keyboard (space/enter when focused). I realise this is unfinished, but I'm filing this now because otherwise it's likely to be missed.

1. Put role="button" on #urlbar-search-icon. As well as exposing the correct accessibility role, that will make it reachable with toolbar keyboard navigation.
2. Support activation with space/enter. I don't know how this thing handles mouse clicks, but it doesn't seem to respond to click (which the toolbar key nav code fires by default), so it'll need to support keypress itself.
3. Give it a tooltip with tooltiptext or an accessibility label with aria-label. A tooltip is more ideal because it makes the purpose clear to users who might not use screen readers but might (for whatever reason) have trouble determining the purpose of the icon. Unless the icon already contains text?

---
🐞 Issue is synchronized with Bugzilla [Bug 1591848](https://bugzilla.mozilla.org/show_bug.cgi?id=1591848)
